src/compose/service-manager.ts

```diff
@@ -32,626 +32,616 @@ type ServiceManagerEventEmitter = StrictEventEmitter<
+	let containerId: string | null = null;
 
-			this.reportNewStatus(containerId, service, 'Starting');
+	try {
+		const container = await create(service);
+		containerId = container.id;
+		logger.logSystemEvent(LogTypes.startService, { service });
 
-			let remove = false;
-			let err: Error | undefined;
-			try {
-				await container.start();
-			} catch (e) {
-				// Get the statusCode from the original cause and make sure it's
-				// definitely an int for comparison reasons
-				const maybeStatusCode = PermissiveNumber.decode(e.statusCode);
-				if (isLeft(maybeStatusCode)) {
-					remove = true;
-					err = new Error(
-						`Could not parse status code from docker error:  ${e}`,
-					);
-					throw err;
-				}
-				const statusCode = maybeStatusCode.right;
-				const message = e.message;
+		reportNewStatus(containerId, service, 'Starting');
 
-				// 304 means the container was already started, precisely what we want
-				if (statusCode === 304) {
-					alreadyStarted = true;
-				} else if (
-					statusCode === 500 &&
-					_.isString(message) &&
-					message.trim().match(/exec format error$/)
-				) {
-					// Provide a friendlier error message for "exec format error"
-					const deviceType = await config.get('deviceType');
-					err = new Error(
-						`Application architecture incompatible with ${deviceType}: exec format error`,
-					);
-					throw err;
-				} else {
-					// rethrow the same error
-					err = e;
-					throw e;
-				}
-			} finally {
-				if (remove) {
-					// If starting the container fialed, we remove it so that it doesn't litter
-					await container.remove({ v: true }).catch(_.noop);
-					logger.logSystemEvent(LogTypes.startServiceError, {
-						service,
-						error: err,
-					});
-				}
+		let shouldRemove = false;
+		let err: Error | undefined;
+		try {
+			await container.start();
+		} catch (e) {
+			// Get the statusCode from the original cause and make sure it's
+			// definitely an int for comparison reasons
+			const maybeStatusCode = PermissiveNumber.decode(e.statusCode);
+			if (isLeft(maybeStatusCode)) {
+				shouldRemove = true;
+				err = new Error(`Could not parse status code from docker error:  ${e}`);
+				throw err;
 			}
-
-			const serviceId = service.serviceId;
-			const imageId = service.imageId;
-			if (serviceId == null || imageId == null) {
-				throw new InternalInconsistencyError(
-					`serviceId and imageId not defined for service: ${service.serviceName} in ServiceManager.start`,
+			const statusCode = maybeStatusCode.right;
+			const message = e.message;
+
+			// 304 means the container was already started, precisely what we want
+			if (statusCode === 304) {
+				alreadyStarted = true;
+			} else if (
+				statusCode === 500 &&
+				_.isString(message) &&
+				message.trim().match(/exec format error$/)
+			) {
+				// Provide a friendlier error message for "exec format error"
+				const deviceType = await config.get('deviceType');
+				throw new Error(
```

Comment:
I just realized the reason for the previous `err =`, it's because you use it in the `finally` so it does need to be assigned (ideally with a comment somewhere about why we're always assigning the err) or the code needs to be reworked to handle it differently